### PR TITLE
fix(@desktop/communities): Computing fees in ETH and USD for Sign Transaction dialog

### PR DESCRIPTION
**Pull request by endulab**
This is a quick pr to show transaction fees for deployment.

TODOs for next milestone:
- make a call async
- UI should decide about formatting, right now Nim returns formatted string

image

### CODE REVIEW

File: ui/app/AppLayouts/Chat/popups/community/SignMintTokenTransactionPopup.qml [hunk 1]
```diff
@@ -19,6 +19,7 @@ StatusDialog {
 
     property string collectibleName
     property string networkName
+    property int chainId
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/popups/community/SignMintTokenTransactionPopup.qml:22]: We don't really need this parameter in the popup context. We can just remove it!

File: ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml [hunk 2]
```diff
@@ -220,10 +220,11 @@ SettingsPageLayout {
                 collectibleName: parent.name
                 accountName: parent.accountName
                 networkName: parent.chainName
+                chainId: parent.chainId
                 feeText: root.feeText
                 isFeeLoading: root.isFeeLoading
 
-                onOpened: root.signMintTransactionOpened()
+                onOpened: root.signMintTransactionOpened(popup.chainId)
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml:227]: ??
```suggestion
                onOpened: root.signMintTransactionOpened(parent.chainId)
```

File: ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml [hunk 3]
```diff
@@ -220,10 +220,11 @@ SettingsPageLayout {
                 collectibleName: parent.name
                 accountName: parent.accountName
                 networkName: parent.chainName
+                chainId: parent.chainId
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml:223]: Not necessary

**Review by noeliaSD — CHANGES_REQUESTED** [review]
Tested and it works fine!

Just added some comments that would be nice to address before integrating. They are easy ones! :)

File: src/app/modules/main/communities/tokens/module.nim [hunk 4]
```diff
@@ -74,3 +77,20 @@ method onUserAuthenticated*(self: Module, password: string) =
     #TODO signalize somehow
   else:
     self.controller.deployCollectibles(self.tempCommunityId, self.tempAddressFrom, password, self.tempDeploymentParams, self.tempTokenMetadata, self.tempChainId)
+
+method computeDeployFee*(self: Module, chainId: int): string =
+  let suggestedFees = self.controller.getSuggestedFees(chainId)
+  if suggestedFees == nil:
+    return "-"
```
[the remark this answers is not in the document]
Comment (endulab) [reply src/app/modules/main/communities/tokens/module.nim:84]: @noeliaSD done

File: ui/app/AppLayouts/Chat/stores/CommunityTokensStore.qml [hunk 5]
```diff
@@ -60,9 +60,9 @@ QtObject {
                                                     infiniteSupply, transferable, selfDestruct, chainId, artworkSource)
     }
 
-    function computeDeployFee() {
-        // TODO: Backend compute minitng fee
-        root.deployFeeUpdated(root.deployFee) // TO BE REMOVED
+    function computeDeployFee(chainId) {
+        communityTokensModuleInst.computeDeployFee(chainId)
+        root.deployFeeUpdated(root.deployFee)
```
[the remark this answers is not in the document]
Comment (endulab) [reply ui/app/AppLayouts/Chat/stores/CommunityTokensStore.qml:66]: Done

**Comment by endulab** [comment]
@noeliaSD I updated the pr.

**Review by noeliaSD — APPROVED** [review]
LGTM! Thanks for the updates!

File: src/app_service/service/community_tokens/service.nim [hunk 6]
```diff
@@ -139,3 +147,12 @@ QtObject:
...
 
+  proc getFiatValue*(self: Service, cryptoBalance: string, cryptoSymbol: string): string =
+    if (cryptoBalance == "" or cryptoSymbol == ""):
+      return "0.00"
+
+    let currentCurrency = self.settingsService.getCurrency()
+
+    let price = self.tokenService.getTokenPrice(cryptoSymbol, currentCurrency)
+    let value = parseFloat(cryptoBalance) * price
+    return fmt"{value:.2f}"
```
Comment (caybro) [inline src/app_service/service/community_tokens/service.nim:158]: How is this supposed to be correctly formatted wrt to decimal separator when we are returning a string here?

Comment (caybro) [reply src/app_service/service/community_tokens/service.nim:158]: This should return a `double` and formatted properly in QML

**Review by caybro — CHANGES_REQUESTED** [review]

File: src/app_service/service/community_tokens/service.nim (hunk 6 above)
Comment (endulab) [reply src/app_service/service/community_tokens/service.nim:158]: There is a ticket for this for next milestone: https://github.com/status-im/status-desktop/issues/10035

Comment (caybro) [reply src/app_service/service/community_tokens/service.nim:158]: Cool, thanks :)

**Review by caybro — APPROVED** [review]

**Pull request merged**